packages/editor/src/components/post-publish-panel/maybe-tags-panel.js

```diff
@@ -34,12 +33,26 @@ const TagsPanel = () => {
 	);
 };
 
-class MaybeTagsPanel extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			hadTagsWhenOpeningThePanel: props.hasTags,
+const MaybeTagsPanel = () => {
+	const { hasTags, isPostTypeSupported } = useSelect( ( select ) => {
+		const postType = select( editorStore ).getCurrentPostType();
+		const tagsTaxonomy = select( coreStore ).getTaxonomy( 'post_tag' );
+		const _isPostTypeSupported = tagsTaxonomy?.types?.includes( postType );
+		const areTagsFetched = tagsTaxonomy !== undefined;
+		const tags =
+			tagsTaxonomy &&
+			select( editorStore ).getEditedPostAttribute(
+				tagsTaxonomy.rest_base
+			);
+		return {
+			hasTags: !! tags?.length,
+			isPostTypeSupported: areTagsFetched && _isPostTypeSupported,
```

Comment:
* it is preferable to derive values like `hasTags` inside `mapSelec`. It will prevent re-rendering components when a new tag is assigned to a post.
* I have combined post-type support check since if tags taxonomy has not been fetched yet, the `_isPostTypeSupported` value will always be `false`